cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,473 @@
+            email='abc@example.com',
+            groups=cb_group,
+            cost_center='Workload',
+            value_assign='Database')
+        vm.set_ownership(user=user.name)
+        logger.info('Assigned VM OWNERSHIP for {} running on {}'.format(vm_name, provider.name))
+
+        yield user.name
+    finally:
+        vm.unset_ownership()
+        if user:
+            user.delete()
+
+
+@pytest.yield_fixture(scope="module")
+def enable_candu(appliance):
+    # C&U data collection consumes a lot of memory and CPU.So, we are disabling some server roles
+    # that are not needed for Chargeback reporting.
+
+    candu = appliance.collections.candus
+    server_info = appliance.server.settings
+    original_roles = server_info.server_roles_db
+    server_info.enable_server_roles(
+        'ems_metrics_coordinator', 'ems_metrics_collector', 'ems_metrics_processor')
+    server_info.disable_server_roles('automate', 'smartstate')
+    candu.enable_all()
+
+    yield
+
+    server_info.update_server_roles_db(original_roles)
+    candu.disable_all()
+
+
+@pytest.yield_fixture(scope="module")
+def assign_custom_rate(new_compute_rate):
+    # Assign custom Compute rate to the Enterprise and then queue the Chargeback report.
+    description = new_compute_rate
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': description}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+    logger.info('Assigning CUSTOM Compute rate')
+
+    yield
+
+    # Resetting the Chargeback rate assignment
+    enterprise = cb.Assign(
+        assign_to="The Enterprise",
+        selections={
+            'Enterprise': {'Rate': '<Nothing>'}
+        })
+    enterprise.computeassign()
+    enterprise.storageassign()
+
+
+def verify_records_rollups_table(appliance, provider):
+    # Verify that hourly rollups are present in the metric_rollups table.
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        if (record.cpu_usagemhz_rate_average or
+           record.cpu_usage_rate_average or
+           record.derived_memory_used or
+           record.net_usage_rate_average or
+           record.disk_usage_rate_average):
+            return True
+    return False
+
+
+@pytest.fixture(scope="module")
+def resource_usage(vm_ownership, appliance, provider):
```

Comment:
**REQUIRED**
This function is huge. Consider to divide it into smaller parts.

Reply:
I've moved verify_records_metrics_table() outside of resource_usage(). That's the best I can do.